libnmstate/nm/sriov.py

```diff
@@ -43,11 +90,67 @@ def create_setting(iface_state, base_con_profile):
         if not sriov_setting:
             sriov_setting = nmclient.NM.SettingSriov.new()
 
+        vfs_config = sriov_config.get(Ethernet.SRIOV.VFS_SUBTREE, [])
+        vf_object_ids = {vf.get_index() for vf in sriov_setting.props.vfs}
+        vf_config_ids = {
+            vf_config[Ethernet.SRIOV.VFS.ID] for vf_config in vfs_config
+        }
+        _edit_sriov_vfs_in_setting(
+            vfs_config, sriov_setting, vf_config_ids & vf_object_ids
+        )
+        _create_sriov_new_vfs_in_setting(
+            vfs_config, sriov_setting, vf_config_ids - vf_object_ids
+        )
+        _remove_sriov_vfs_in_setting(
+            vfs_config, sriov_setting, vf_object_ids - vf_config_ids
+        )
+
         sriov_setting.props.total_vfs = sriov_config[Ethernet.SRIOV.TOTAL_VFS]
 
     return sriov_setting
 
 
+def _create_sriov_new_vfs_in_setting(vfs_config, sriov_setting, vf_ids_to_add):
+    vfs_config_to_add = (
+        vf_config
+        for vf_config in vfs_config
+        if vf_config[Ethernet.SRIOV.VFS.ID] in vf_ids_to_add
+    )
+    for vf_config in vfs_config_to_add:
+        vf_object = nmclient.NM.SriovVF.new(vf_config[Ethernet.SRIOV.VFS.ID])
+        for key, val in vf_config.items():
+            _set_nm_attribute(vf_object, key, val)
+
+        sriov_setting.add_vf(vf_object)
```

Comment:
This is now mutable and we can make it immutable by making this function a generator:
```
yield vf_object
```
Then, doing something like this at the caller:
```
for vf_object in _create_sriov_new_vfs_objects():
    sriov_setting.add_vf(vf_object)
```

Reply:
Done in create and remove functions :-)